### ISSUE TITLE
Inject entries into the service catalog

### ISSUE THREAD

**Issue by lvh** [rackerlabs/mimic#219]
Two use cases:

1. I am writing tests for something that orchestrates APIs but don't want to hit production Nova. I hook otter up to Mimic, then tell Mimic about that instance so that it shows up in the service catalog.
2. I am writing things that test some other complex orchestration system. I want to run all my tests, but not everything has Mimic mocks yet. I don't want to hard code the things that don't use Mimic; my tests should just grab things from the service catalog. Injecting entries into the service catalog means that I can start introducing more and more Mimic mocks piecemeal as necessary.

Ideally this would eventually happen by copying the identity admin API.

**Comment by lekhajee** [comment]
Link to the identity admin API to add endpoints to a service catalog: http://developer.openstack.org/api-ref-identity-v2-ext.html#os-kscatalog-ext

**Comment by BenjamenMeyer** [comment]
I'm basically working this now for Cloud Backup; taking over from @lvh

**Comment by BenjamenMeyer** [comment]
PRs #608 and #614 are steps towards this.

**Comment by BenjamenMeyer** [comment]
I'm presently only addressing external APIs; but we could probably do something similar for internal APIs too and therefore address #230 in the process. Let's get the external API work done first.

**Comment by BenjamenMeyer** [comment]
Once #614 is merged, then I'll submit another PR with all the end-points for the OS-KSCATALOG extension (/endpointTemplates and /tenantid/endpointPointeTemplates) and a few endpoints for OS-KSADM extension (/services).
